pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -5552,20 +5553,38 @@ metadata:
 	want.Status = v1beta1.PipelineRunStatus{
 		PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
 			PipelineSpec: ps.DeepCopy(),
+			Provenance: &v1beta1.Provenance{
+				ConfigSource: &v1beta1.ConfigSource{
+					URI: "abc.com",
+					Digest: map[string]string{
+						"sha1": "a123",
+					},
+					EntryPoint: "foo/bar",
+				},
+			},
 		},
 	}
 	want.ObjectMeta.Labels["tekton.dev/pipeline"] = pr.ObjectMeta.Name
 
 	// The first time we set it, it should get copied.
-	if err := storePipelineSpecAndMergeMeta(pr, &ps, &pr.ObjectMeta); err != nil {
+	if err := storePipelineSpecAndMergeMeta(pr, &ps, &tresources.ResolvedObjectMeta{
```

Comment:
So according to my comment in pipelinerun_test.go, if storePipelineSpecAndMergeMeta is called twice once without the ResolvedObject and 2nd time with the resolvedObjectMeta then the ConfigSource should not get stored.
Either this is expected or not, please add a test in either case as a documentation of the behavior.

Reply:
Thanks @jagathprakash. SG. Added the tests. PTAL!